server/graphql/mutations/orders.js

```diff
@@ -368,4 +368,24 @@ export function updateSubscription(remoteUser, args) {
     return updatePromise
         .then(() => order.update({ PaymentMethodId: newPm.id}))
   })
-}
\ No newline at end of file
+}
+
+export async function refundTransaction(_, args, req) {
+  // 0. Retrieve transaction from database
+  const transaction = await models.Transaction.findById(args.id, {
+    include: [models.Order, models.PaymentMethod]
+  });
+
+  // 1. Verify user permission. User must be either
+  //   a. User that created transaction (within 24h) -- Not implemented yet
+  //   b. Host or Admin of the collective receiving the donation
+  if (req.remoteUser.id != transaction.CreatedByUserId &&
+      !req.remoteUser.hasRole([roles.ADMIN, roles.HOST],
+                              transaction.CollectiveId)) {
+    throw new errors.Unauthorized({ message: "Not an admin neither owner" });
+  }
```

Comment:
Hi @asood123, should I keep both HOST & ADMIN here or just ADMIN for now? I don't quite remember what we ended up deciding.

Reply:
Is plan to release to all admins or internally for us first?

Reply:
Hmm, well It'd be certainly nice if we could test it once and see how it looks first! I'll get rid of the `ADMIN` for now!

Reply:
I meant getting rid of the `HOST` flag and keep it to "Admins" first. :kick_scooter:

Reply:
Well, then any admin could do it right? If we want to limit it to internal, you might need a new function

Reply:
Oh thanks for clarifying I totally confused the roles here :/ I'll change it!! Thanks for the patience!!